lib/modules/datasource/deno/index.ts

```diff
@@ -0,0 +1,109 @@
+import is from '@sindresorhus/is';
+import pMap from 'p-map';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { regEx } from '../../../util/regex';
+import { joinUrlParts } from '../../../util/url';
+import * as semanticVersioning from '../../versioning/semver';
+import { Datasource } from '../datasource';
+import type { Release } from '../index';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import type {
+  DenoAPIModuleResponse,
+  DenoAPIModuleVersionResponse,
+} from './types';
+import { createSourceURL, tagsToRecord } from './utils';
+
+export class DenoDatasource extends Datasource {
+  static readonly id = 'deno';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = semanticVersioning.id;
+
+  override readonly defaultRegistryUrls = ['https://apiland.deno.dev'];
+
+  constructor() {
+    super(DenoDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${DenoDatasource.id}`,
+    key: ({ packageName, registryUrl }: GetReleasesConfig) =>
+      // TODO: types (#7154)
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      `${registryUrl}:${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const massagedRegistryUrl = registryUrl ?? this.defaultRegistryUrls[0];
+
+    const extractResult = regEx(
+      '^(https:\\/\\/deno.land\\/)(?<rawPackageName>[^@\\s]+)'
```

Comment:
Is this still the case when a non default registry is used?

Reply:
This datasource is only for `deno.land` other registry have different APIs ( e.g. `nest.land`) .    
The `registryUrl` can be used to host a custom proxy of the API server.

Reply:
Custom registry support is true though? Is that for people who just set up a proxy to deno land

Reply:
Yes, my intend has been to allow to define proxies for deno land